New home page integration

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR replaces #743 and includes:

- Ama's original changes
- Nik's revisions

The changes to the original PR kept growing and rebasing the old branch onto `main` proved too time consuming. This is a clean PR with just the changed files.

[**LIVE PREVIEW**](https://docs-demo.vm.validmind.ai/pr_previews/ama8636/sc-10590/implement-new-homepage-design/index.html)

## Nik's revisions

**`index.qmd`**

- Completed the implementation of the search box design 
- Adjusted hero banner padding to better match the design
- Refactored section padding for closer alignment with the design, for example "Setup + Administration" no longer wraps
- Fixed mobile responsiveness glitches
- Switched to Tachyons CSS where available to replace new CSS that duplicated it
- Added a fade effect around the background in the ValidMind Tour section to avoid a cut-off look and fixed width 
- Updated release notes content to reflect the latest April release

**`styles.css`**

- Restored the missing search icon, previously hidden everywhere, now hidden only on the home page
- De-duplicated CSS styles, for example we had MULTIPLE definitions for smallcaps and buttons; more is needed
- Removed unused CSS from the old home page, also in `index.qmd`
- Consolidated CSS specific to the home page, such as styles for the tour section  

**Footer**

- Recombined footer content for a cleaner layout, previously docs site links where scattered across left, center, and right which looked slightly odd:
   - `.nav-footer-left` — company stuff, cookie preferences
   - `.nav-footer-center` — all the docs-site links
   - `.nav-footer-right` — just flex box white space
- Removed dividers and replaced with better whitespace
- Fixed the footer alignment on the home page so that it matches the design, previously you had to scroll to see it

## Open issues:

**CSS**

- [x] MAJOR: Review all CSS changes, there are many additions in `styles.css` that still need to be reviewed ... I removed a couple hundred lines of new CSS already
- [ ] Perform additional mobile responsive testing 
- [x] Address heading styles, current options don’t match the boldness used in the design; update globally  > [sc-11237](https://app.shortcut.com/validmind/story/11237)
- [x] Figure out font sizing, many of our fonts are in-between what the design uses. For example, `.f5` is too small and `.f4` is too large, for example > [sc-11237](https://app.shortcut.com/validmind/story/11237)
- [x] Move the chatbot above the footer; I pushed some experimental CSS that seems to work locally, but there might be a better option [solved by better white space handling and removing some company links]

**Content**
- [ ] This PR contains the fake product tour from Beck's old PR, need to replace
- [x] Decide on renaming the “Configuration” section to “Setup + Administration”  [Decision is to undo the renaming]

## Screenshots

**Figma design**

<img width="1471" alt="image" src="https://github.com/user-attachments/assets/26670d43-6957-482a-b94b-2289576fb0ea" />

**Nik's revisions**

<img width="1564" alt="image" src="https://github.com/user-attachments/assets/25367ac8-f6aa-42db-a450-452ffc465521" />

**Ama's original implementation** 

<img width="1471" alt="image" src="https://github.com/user-attachments/assets/e3285532-5587-46d9-a207-50806db1cecb" />

Note the footer is visible only when scrolling:

![Capto_ 2025-06-18_08-06-09_am2](https://github.com/user-attachments/assets/c4e7620b-e645-48bb-ba66-57f493261150)


<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `breaking-change`
- `deprecation`
- `bug`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

Our docs site has a fresh design with a new home page that makes it easier to get started with ValidMind. [Try it!](https://docs.validmind.ai)

<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

validbeck reviewed (changes requested):
I'll try to comment as I find the issues. So far on first glance:

## Homepage

### Navbar

<img width="1719" alt="Screenshot 2025-06-23 at 10 26 54 AM" src="https://github.com/user-attachments/assets/2b1e18db-88a9-4968-84df-b04ba038b0ba" />

I can't remember if this was in the design (probably was) but I am *not* in favour of flattening the navigation here on the home page. I think the *homepage* should have links to all the subpages, but the *subpages* should have the minimized drop-down menu.

Here's my logic:

As a new user, if I land on a docs site and have to dig around for the section I need (not readily apparently just from base navigation), that's frustrating and confusing.

Whereas in a subsection, I am already looking at a targeted topic and am likely more interested in the side/sub-navigation, so hiding the top-level navigation is acceptable.

Scrolling all the way down to the footer to get what I need should be a *last resort*, not a first choice.

### Search box

There's some wacky squished scrollbar situation going on here that makes it both ugly and hard to navigate, we might need to make the box wider to accommodate:

| Old | New |
|--|--|
| <img width="1040" alt="Screenshot 2025-06-23 at 10 24 26 AM" src="https://github.com/user-attachments/assets/1e37fec4-48f8-4b89-bb76-76688e74f873" /> | ![Screenshot 2025-06-23 at 10 22 53 AM](https://github.com/user-attachments/assets/5fd25fd9-e052-4c40-a211-7e58a029799e)|

### Product Tour

https://github.com/user-attachments/assets/700ae2a5-7ccb-4dbb-be33-6c10d8f0f3a4

**lol**

this doesn't exist yet, so maybe we shouldn't link to it? Or at least have a better placeholder? 

### Latest Release

The alignment for the announcement and the actual line of release version / date bothers me — it looks like it's superscripted for no reason:

![Screenshot 2025-06-23 at 10 20 40 AM](https://github.com/user-attachments/assets/0df1ce6d-41b9-4349-b6b7-b408aa503ffe)

### Footer

The footer text is WAY too small. This is just on my 27" with 2k, so this isn't acceptable. 

![Screenshot 2025-06-23 at 10 20 49 AM](https://github.com/user-attachments/assets/2fdc4364-2945-4393-b39e-85bb57eb6b80)

There's also some weirdness with the `What's new?` link that somehow looks different than the other links? Speaking of links ...

These are all broken. What's up with that?

<img width="337" alt="Screenshot 2025-06-23 at 10 29 56 AM" src="https://github.com/user-attachments/assets/22b10016-fd89-4c2a-bfac-57abe44dfb00" />

And what's with the repeat "Get started" and "Onboarding" links? Why does "Onboarding" link to Manage validation guidelines? 🤔 

<img width="172" alt="Screenshot 2025-06-23 at 10 30 37 AM" src="https://github.com/user-attachments/assets/36bb50fa-d2d6-46cf-9e05-f9297d0dad80" />

Why does "Model inventory" link to Guides? Shall we just change this link to Guides?

... actually, all the links in the footer are, respectfully, a hot mess. I don't know why we choose those links in that order. Can someone explain?

All the ValidMind Academy links seem to link to the same top-level, for example...

## Subpages

### Developer portal

The "Login" button on the Developer 'portal' is too jarring — it worked with the old theme but not this one:

| Old | New |
|--|--|
| <img width="865" alt="Screenshot 2025-06-18 at 1 11 01 PM" src="https://github.com/user-attachments/assets/74b3f085-9d02-4608-bd60-01d6388c58a6" />| <img width="857" alt="Screenshot 2025-06-23 at 10 14 53 AM" src="https://github.com/user-attachments/assets/8ddebc7d-8540-437a-93bc-7489a961589f" /> |

We might just need to quickly patch `developer.css` in the interim for this. Speaking of... I patched the padding issue in another PR, but the lighter green buttons don't work with the portal layout — might be another quick workaround we need to implement.

<img width="865" alt="Screenshot 2025-06-18 at 1 11 01 PM" src="https://github.com/user-attachments/assets/ea00d88b-9ae4-49ad-b0c5-59c72693b6b0" />

### Training

Are the training FAQs _supposed_ to be like this (no border around the buttons, text reverted from smallcaps)? 

| Old | New |
|--|--|
| <img width="1723" alt="Screenshot 2025-06-23 at 10 18 18 AM" src="https://github.com/user-attachments/assets/dcd3a4dd-d66a-4b5d-91c3-b2f2e4216549" />| <img width="1725" alt="Screenshot 2025-06-18 at 1 11 46 PM" src="https://github.com/user-attachments/assets/0e318a3e-7e8a-4817-9416-2a3801876cae" /> |

What happened to our friend `.smallercaps`? And again, are we reverting the buttons away from `.smallcaps`?

| Old | New |
|--|--|
| <img width="1162" alt="Screenshot 2025-06-23 at 10 37 03 AM" src="https://github.com/user-attachments/assets/02d0870b-e10e-4d77-b643-86c5382fd175" />| <img width="1143" alt="Screenshot 2025-06-23 at 10 35 54 AM" src="https://github.com/user-attachments/assets/6b859260-401f-4d5c-a3f6-801dcc0a0ee2" />| 

## Mobile responsiveness

| Uh | Oh |
|--|--|
| ![Screenshot 2025-06-23 at 10 39 05 AM](https://github.com/user-attachments/assets/a2a54cdc-5c45-4659-bf24-a23b3c3b4764) | ![Screenshot 2025-06-23 at 10 39 49 AM](https://github.com/user-attachments/assets/a0a7273b-3e4b-41f4-814d-60edf2937c16)|

I honestly don't know how this happened and therefore I don't know how to fix it 💀

nrichers commented:
@LoiAnsah @validbeck notes with to-dos from our session just now ...

## Beck's feedback

> I can't remember if this was in the design (probably was) but I am not in favour of flattening the navigation here on the home page. I think the homepage should have links to all the subpages, but the subpages should have the minimized drop-down menu.

Let's implement the design, Nik to open a follow-on story to do user testing on our navigation and fix site-wide if necessary.

> There's some wacky squished scrollbar situation going on here that makes it both ugly and hard to navigate, we might need to make the box wider to accommodate

Nik to look up the code and attempt a fix, share with Beck; Beck to verify.

> Product Tour

Nik to ask Kevin if we have a product tour video we can use for now. 

> The alignment for the announcement and the actual line of release version / date bothers me 

Ama to switch back to the listing tile and then fix to the CSS. 

Nik to provide a link to a pre-filtered highlights page and provide to Ama. [The link will be `/releases/feature-highlights.qmd#category=highlight` but Nik needs to create this page.] 

> The footer text is WAY too small.

Ama to experiment with font sizes and use a uniform font size across the entire footer. 

> Speaking of links ...
> These are all broken. What's up with that?
> Why does "Model inventory" link to Guides?
>All the ValidMind Academy links seem to link to the same top-level

Ama to go through the footer links, find the correct link targets, and update them to point to the correct pages. Also, there is a casing issue for the model inventory link (should be "Model inventory").

> There's also some weirdness with the What's new? link 

Ama to remove the link.

> And what's with the repeat "Get started" and "Onboarding" links? 

Ama to change "Onboarding" back to "Configuration" and link to what is currently called "Onboarding" on the "Guides" landing page. That heading ("Onboarding") also needs to be changed to "Configuration".

> The "Login" button on the Developer 'portal' is too jarring

Ama to go back to the old button by updating the developer CSS to override that button. You need to update `site/developer/developer.css` with the old CSS, please ask if you want help.

> Are the training FAQs supposed to be like this (no border around the buttons, text reverted from smallcaps)?
> What happened to our friend .smallercaps? And again, are we reverting the buttons away from .smallcaps?

Nik to check why the CSS for `.smallercaps` and `.button-small` was removed and re-add it.

> Mobile responsiveness
>I honestly don't know how this happened and therefore I don't know how to fix it 💀

Ama to check the mobile view (be sure to switch your browser to the correct view) and figure out why the content is smooshed to one side. 

## Nik's open issues

>  Decide on renaming the “Configuration” section to “Setup + Administration” 

Undo the renaming, should be "Configuration".

_MORE TO COME ..._

LoiAnsah commented:
###  **Completed Tasks**
- Footer links:
All links reviewed.
Correct targets found and updated.
Casing issue with “Model inventory” fixed.

- “What’s new?” link:
Link removed.

- "Onboarding" duplicate issue:
“Onboarding” changed back to “Configuration”.

- Link updated to target the current “Onboarding” page on the Guides landing page.
Heading on the target page also changed to “Configuration”.

- Changed the size of the footer writing, need someone to test it too. On my screen, it looks fine.

### ** Still To Do**
Listing tile + CSS Fixes:

Ama needs to switch back to the listing tile layout and apply appropriate CSS fixes.

- Login button style: The Current styling on the Developer portal is too jarring. Revert to old button styling by updating site/developer/developer.css (ask for help if needed).

- Mobile responsiveness: The current mobile layout is broken or squished, inspect using mobile view in browser dev tools and fix the content layout.

nrichers commented:
One small update from Nik: 
 
> Let's implement the design, Nik to open a follow-on story to do user testing on our navigation and fix site-wide if necessary.

Opened [User testing for flattened navigation on home page (sc-11209)](https://app.shortcut.com/validmind/story/11209).

validbeck commented:
Search bar is even tinier, not sure how or why 🤔 

<img width="1707" height="907" alt="Screenshot 2025-07-10 at 3 13 47 PM" src="https://github.com/user-attachments/assets/5acc6fe4-402e-4d38-bbf0-63793129f1fb" />

nrichers commented:
OK, searchbox should be fixed, though I will need @validbeck to confirm that the scrollbar issue is resolved as well:

<img width="1508" height="968" alt="Capto_ 2025-07-10_05-12-02_pm" src="https://github.com/user-attachments/assets/97438bfa-5626-4a5a-8384-63d7bf778b3c" />

@LoiAnsah 949fb737706b60a56703bd88fba9a8c19003927b removed the minimum width from the searchbox. I basically just re-added it and then made the box slightly wider to (hopefully) fix the scrollbar issue. If Beck reports that the issue still persists, I can make the box wider still.

nrichers commented:
I made one more footer improvement for non-mobile users:

**Minimum width footer display**

<img width="1000" height="634" alt="image" src="https://github.com/user-attachments/assets/638e29c3-95f4-4ce9-8ac9-15d18a62b62f" />

**Below minimum width** 

<img width="964" height="634" alt="image" src="https://github.com/user-attachments/assets/a4a11033-7166-42cf-89ec-3273f6765841" />

nrichers commented:
@LoiAnsah I fixed the responsive footer issue you had shown me and did a bit more CSS cleanup. 

This fix was a combination of:

- HTML improvements to put each line under `footer-company-info` into its own div
- CSS additions and cleanup to ensure we control the display of these lines consistently

**Before** (screenshot from Zoom)

<img width="504" height="301" alt="Capto_ 2025-07-14_12-51-46_pm2" src="https://github.com/user-attachments/assets/68323e50-2f2d-4ca5-95db-60db6f51c5cf" />

**After** (responsive)

<img width="497" height="259" alt="Capto_ 2025-07-15_05-35-40_pm" src="https://github.com/user-attachments/assets/302c8238-278b-40d6-89d3-69f4c0d3480e" />

**After** (mobile)

<img width="402" height="253" alt="Capto_ 2025-07-15_05-34-38_pm" src="https://github.com/user-attachments/assets/5a292b47-669b-42ff-a169-f07d9d38148c" />

I also fixed a display issue with the slash in the searchbox, hiding it below a certain size.

validbeck commented on site/index.qmd:
```diff
@@ -129,276 +137,407 @@ a:hover {
 
-:::{#validmind-academy}
+::: {.f3 .b}
+ValidMind [Tour]{.pink}
```
GLOBAL COMMENT: Not too important since we're just trying to get the MVP out, but we don't seem to be using the product variables anywhere on this index page. We should fix that in the next iteration.

nrichers replied:
Opened https://app.shortcut.com/validmind/story/11339/switch-new-home-page-to-use-product-variables to track this fix.

validbeck commented on site/index.qmd:
```diff
@@ -129,276 +137,407 @@ a:hover {
-::::
-
+::: {.nt3}
+[**What is ValidMind?**](https://validmind.com/request-demo/){.button .button-white}
```
```suggestion
[**Request a Demo**](https://validmind.com/request-demo/){.button .button-white}
```

validbeck replied:
As a user, I'd be kind of annoyed if I clicked on a page that promised some explanatory blurb just to be taken to a sales demo request form. Let's be transparent here while we use this placeholder.

validbeck commented on site/index.qmd:
```diff
@@ -129,276 +137,407 @@ a:hover {
-[[all releases {{< fa chevron-right >}} ](releases/all-releases.qmd)]{.smallcaps}
+::: {.pl4 .mid-gray}
 
+::: {#index-releases}
```
Is this the way the newest release is meant to look? We also seem to have gotten rid of the "View all releases" link?

<img width="638" height="300" alt="Screenshot 2025-07-17 at 4 51 07 PM" src="https://github.com/user-attachments/assets/e37d7e98-38a0-40a9-b130-c8eb346e9478" />

nrichers replied:
This is now fixed, along with better display of the latest release features:

<img width="659" height="358" alt="Capto_ 2025-07-17_05-45-14_pm2" src="https://github.com/user-attachments/assets/a32d9067-7cce-47f1-af28-a5f7223e038f" />

Is responsive and also works on mobile:

<img width="403" height="590" alt="Capto_ 2025-07-17_05-45-44_pm2" src="https://github.com/user-attachments/assets/9b9e48d4-e1f6-48e9-8a51-0602649ce613" />

validbeck reviewed (changes requested):
I am not sure when this was introduced, but we're missing the alternative logo from the developer "portal" in the top-left of the navbar:

<img width="1363" height="1327" alt="Screenshot 2025-07-17 at 4 45 46 PM" src="https://github.com/user-attachments/assets/3189ecfe-9d8d-45a9-bdfe-098867995403" />

[It was still here when I first reviewed this PR](https://github.com/validmind/documentation/pull/764#pullrequestreview-2950908024), so it must have been a subsequent change. 

Since it takes users back to the homepage, I think it's important enough to need to be added back in — pages also look strange without a logo at the top corner.

I wish I could approve this as is but I think a missing logo and the missing releases link are important.

nrichers commented:
OK, I figured out the whitespace issue we were having with the ValidMind Academic and Latest Features row, fixed. This is what the page now looks like: 

<img width="1552" height="1458" alt="image" src="https://github.com/user-attachments/assets/dd01e390-b7d3-44e4-bef7-aa6d06b8273b" />

Also, this took me three hours to troubleshoot ... and I think I need to pick @validbeck's brain further:

- Even flex has some responsiveness issues unless you are _very_ careful with `@media` `max-width`. The filler columns do NOT guarantee a responsive page, see this example: 

   <img width="601" height="516" alt="Capto_ 2025-07-17_09-22-55_pm" src="https://github.com/user-attachments/assets/da4b9ae7-cb0a-42f3-81f9-4e73bc6e27b5" />

- The .home-section-card CSS @LoiAnsah introduced is nicely responsive and doesn’t require the filler column workaround we’ve used with flex, so I removed those columns again, as they were creating excessive whitespace on the left and right at the top of the page. I also moved the CSS into index.qmd since it’s not relevant to the entire site.

-  The current page still has slightly more whitespace on the left and right edges of the first and last row compared to the Figma design. That said, it looks much clearer to me and we gained the room for that whitespace by reducing the gradient padding on both sides.

Let’s age this on staging for a couple of weeks and handle other changes in separate PRs.
